Modules/getpath.c

```diff
@@ -82,27 +82,32 @@ getpath_abspath(PyObject *Py_UNUSED(self), PyObject *args)
 static PyObject *
 getpath_basename(PyObject *Py_UNUSED(self), PyObject *args)
 {
-    const char *path;
-    if (!PyArg_ParseTuple(args, "s", &path)) {
+    PyObject *path;
+    if (!PyArg_ParseTuple(args, "U", &path)) {
```

Comment:
BTW, I would use `METH_O` and `PyArg_Parse()` in these functions, but this is another issue.

Why cannot they be implemented in Python?

Reply:
> BTW, I would use METH_O and PyArg_Parse() in these functions, but this is another issue.

I tried to minimize the changes.

> Why cannot they be implemented in Python?

Ask @zooba who designed this. Maybe it can be changed?